test/e2e/specs/jetpack/social__editor-features.ts

```diff
@@ -161,9 +188,30 @@ describe( DataHelper.createSuiteTitle( 'Social: Editor features' ), function ()
 				const toggle = section.getByLabel( 'Share when publishing' );
 				expect( await toggle.isVisible() ).toBe( false );
 
-				// Verify whether the resharing button is visible.
-				reshareButton = section.getByRole( 'button', { name: 'Share post', exact: true } );
-				expect( await reshareButton.isVisible() ).toBe( features.resharing );
+				// Check if the Preview & Share button is visible based on resharing feature
+				sharePostModalButton = section.getByRole( 'button', {
+					name: 'Preview & Share',
+					exact: true,
+				} );
+
+				expect( await sharePostModalButton.isVisible() ).toBe( features.resharing );
```

Comment:
If resharing is not available we do not show the modal in the published post screen